buildengine/engine.go

```diff
@@ -285,12 +285,19 @@ func (e *Engine) watchForModuleChanges(ctx context.Context, period time.Duration
 				delete(e.projects, config.Key)
 			case WatchEventProjectChanged:
 				config := event.Project.Config()
+
+				lastBuildTime := e.projects[config.Key].GetLastBuildStartTime()
```

Comment:
I think rather than making all these changes because of making Project mutable, just change the `e.projects` map to use an intermediate struct that contains the `Project` + `time.Time`:

```go
type projectMeta struct {
  project Project
  lastBuildStartTime time.Time
}
```

Reply:
Love it! Thanks dude!